### PR TITLE
General IP intel tweaks

### PR DESCRIPTION
**Pull request by optimumtact**

### CODE REVIEW

**Comment by optimumtact** [comment]
This should be safe to test merge without ill effect.

File: code/controllers/subsystem/ipintel.dm [hunk 1]
```diff
@@ -178,6 +177,8 @@ SUBSYSTEM_DEF(ipintel)
 	return TRUE
 
 /datum/controller/subsystem/ipintel/proc/is_exempt(client/player)
+	if(player.holder)
+		return TRUE
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:181]:
```suggestion
	if(player.holder || GLOB.deadmins[player.ckey])
		return TRUE
```

File: code/controllers/configuration/entries/general.dm [hunk 2]
```diff
@@ -453,7 +453,7 @@
 /datum/config_entry/string/ipintel_email
 
 /datum/config_entry/string/ipintel_email/ValidateAndSet(str_val)
-	return str_val != "ch@nge.me" && ..()
+	return str_val != "ch@nge.me" && (!length(str_var) || findtext(str_val, "@")) && ..()
```
Comment (MrStonedOne) [inline code/controllers/configuration/entries/general.dm:456]:
```suggestion
	return str_val != "ch@nge.me" && (!length(str_val) || findtext(str_val, "@")) && ..()
```

File: code/controllers/subsystem/ipintel.dm [hunk 3]
```diff
@@ -30,47 +18,39 @@ SUBSYSTEM_DEF(ipintel)
...
-/datum/controller/subsystem/ipintel/Initialize()
+/datum/controller/subsystem/ipintel/OnConfigLoad()
 	var/list/fail_messages = list()
-
-	probability_threshold = CONFIG_GET(number/ipintel_rating_bad)
-	if(probability_threshold < 0 || probability_threshold > 1)
-		fail_messages += list("invalid probability threshold")
-
+	
 	contact_email = CONFIG_GET(string/ipintel_email)
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:24]:
```suggestion
	var/contact_email = CONFIG_GET(string/ipintel_email)
```

File: code/controllers/subsystem/ipintel.dm [hunk 4]
```diff
@@ -109,17 +89,18 @@ SUBSYSTEM_DEF(ipintel)
 	if(!initialized)
 		return IPINTEL_UNKNOWN_INTERNAL_ERROR
 
-	rate_limits[IPINTEL_RATE_LIMIT_MINUTE] += 1
-	rate_limits[IPINTEL_RATE_LIMIT_DAY] += 1
+	rate_limit_minute += 1
 
-	var/query_base = "https://[src.query_base]/check.php?ip="
+	var/query_base = "https://[CONFIG_GET(string/ipintel_base)]/check.php?ip="
 	var/query = "[query_base][address]&contact=[contact_email]&flags=b&format=json"
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:95]:
```suggestion
	var/query = "[query_base][address]&contact=[CONFIG_GET(string/ipintel_email)]&flags=b&format=json"
```

File: code/controllers/subsystem/ipintel.dm [hunk 5]
```diff
@@ -150,13 +134,17 @@ SUBSYSTEM_DEF(ipintel)
 	qdel(query)
 
 /datum/controller/subsystem/ipintel/proc/fetch_cached_ip_intel(address)
-	var/date_restrictor
-	if(ipintel_cache_length > 0)
-		date_restrictor = " AND date > DATE_SUB(NOW(), INTERVAL [ipintel_cache_length] DAY)"
+	if (!SSdbcore.Connect())
+		return
+	ipintel_cache_length = CONFIG_GET(number/ipintel_cache_length)
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:139]:
```suggestion
	var/ipintel_cache_length = CONFIG_GET(number/ipintel_cache_length)
```

File: code/controllers/subsystem/ipintel.dm [hunk 6]
```diff
@@ -199,9 +190,13 @@ SUBSYSTEM_DEF(ipintel)
 		qdel(query)
 		return FALSE
 	query.NextRow()
-	return !!query.item // if they have a row, they are whitelisted
+	. = !!query.item // if they have a row, they are whitelisted
+	qdel(query)
+	
 
 ADMIN_VERB(ipintel_allow, R_BAN, "Whitelist Player VPN", "Allow a player to connect even if they are using a VPN.", ADMIN_CATEGORY_IPINTEL, ckey as text)
+	if (!is_enabled())
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:198]:
```suggestion
	if (!SSipintel.is_enabled())
```

File: code/controllers/subsystem/ipintel.dm [hunk 7]
```diff
@@ -224,6 +219,8 @@ ADMIN_VERB(ipintel_allow, R_BAN, "Whitelist Player VPN", "Allow a player to conn
 	message_admins("IPINTEL: [key_name_admin(user)] has whitelisted '[ckey]'")
 
 ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a player's VPN whitelist.", ADMIN_CATEGORY_IPINTEL, ckey as text)
+	if (!is_enabled())
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:222]:
```suggestion
	if (!SSipintel.is_enabled())
```

File: code/controllers/subsystem/ipintel.dm [hunk 8]
```diff
@@ -238,6 +235,8 @@ ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a playe
 	message_admins("IPINTEL: [key_name_admin(user)] has revoked the VPN whitelist for '[ckey]'")
 
 /client/proc/check_ip_intel()
+	if (!is_enabled())
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:238]:
```suggestion
	if (!SSipintel.is_enabled())
```

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
Missed some things, made some typos.

**Comment by MrStonedOne** [comment]
Set living to 5 hours, it complained about an error on the one person who had less than that but didn't check anybody else. likely we got the outgoing ip blacklisted, easy fix since i have 5 of them. anyways the rest of the code works so the todo return can be removed.

File: code/controllers/subsystem/ipintel.dm [hunk 9]
```diff
@@ -273,9 +272,11 @@ ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a playe
 			else
 				message_admins("IPINTEL: [key_name_admin(src)] was unable to be checked due to an error.")
 
+	// TODO remove after test merge
+	return
 	if(!connection_rejected)
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:277]:
```suggestion
				message_admins("IPINTEL: [key_name_admin(src)] was unable to be checked due to an error.")

	if(!connection_rejected)
```

**Review by ZephyrTFA — APPROVED** [review]

File: code/controllers/subsystem/ipintel.dm [hunk 10]
```diff
@@ -1,26 +1,14 @@
...
-	/// The length of time (days) to cache IP intel
-	var/ipintel_cache_length
-	/// The living playtime (minutes) for players to be exempt from IPIntel checks
-	var/exempt_living_playtime
 
 	/// Cache for previously queried IP addresses and those stored in the database
 	var/list/datum/ip_intel/cached_queries = list()
 	/// The store for rate limiting
-	var/list/rate_limits
+	var/list/rate_limit_minute = list()
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:11]:
```suggestion
	var/list/rate_limit_minute
```

**Comment by MrStonedOne** [comment]
Ok no, its not the ratelisting. I was able to simulate a query in chrome on sybil's vm and it ran just fine, but so far its not successfully making queries in game.

We'll need to add some debug logging to the actual requests.

File: code/controllers/subsystem/ipintel.dm [hunk 11]
```diff
@@ -109,17 +89,18 @@ SUBSYSTEM_DEF(ipintel)
 	if(!initialized)
 		return IPINTEL_UNKNOWN_INTERNAL_ERROR
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:91]:
```suggestion

```

**Comment by optimumtact** [comment]
> Ok no, its not the ratelisting. I was able to simulate a query in chrome on sybil's vm and it ran just fine, but so far its not successfully making queries in game.
> 
> We'll need to add some debug logging to the actual requests.

did my debug log not work.

**Comment by MrStonedOne** [comment]
it wasn't initializing so it never reached the debug line, which is how i figured out what it was

File: code/controllers/subsystem/ipintel.dm (hunk 10 above)
Comment (MrStonedOne) [reply code/controllers/subsystem/ipintel.dm]: this one was my fuck up

**Comment by MrStonedOne** [comment]
now the debug logs are printing.

Its a blank request body. Likely gonna need to see the http status code to understand. Maybe even the response headers.

File: code/controllers/subsystem/ipintel.dm [hunk 12]
```diff
@@ -92,34 +72,32 @@ SUBSYSTEM_DEF(ipintel)
...
-	var/query_base = "https://[src.query_base]/check.php?ip="
-	var/query = "[query_base][address]&contact=[contact_email]&flags=b&format=json"
+	var/query_base = "https://[CONFIG_GET(string/ipintel_base)]/check.php?ip="
+	var/query = "[query_base][address]&contact=[CONFIG_GET(string/ipintel_email)]&flags=b&format=json"
 
 	var/datum/http_request/request = new
 	request.prepare(RUSTG_HTTP_METHOD_GET, query)
 	request.execute_blocking()
 	var/datum/http_response/response = request.into_response()
 	var/list/data = response.body
```
Comment (MrStonedOne) [inline code/controllers/subsystem/ipintel.dm:98]:
```suggestion
	var/list/data = json_decode(response.body)
```

**Comment by jlsnow301** [comment]
does this close #83008

**Comment by MrStonedOne** [comment]
> does this close #83008

Likely, i added some db status checks to ipintel but would need somebody to test first

**Pull request merged**